doc/tutorial/hello_world.dox

```diff
@@ -0,0 +1,165 @@
+ * matches any single topic level, and a @c # matches all topic levels
+ * from that level on.  In this instance since the publisher is allowing
+ * wildcard matching, the subscriber could use either @c a/b/+/d or @c
+ * a/# (among others) as the topic and still receive the publication.
+ *
+ * A subscription may also be created with multiple topics.  The
+ * publication must include @em all of the topics to be received.
+ *
+ * @see DPS_SetSubscriptionData(), DPS_GetSubscriptionData(),
+ *      DPS_SubscriptionGetNode()
+ *
+ * @subsection receiving-a-publication Receiving a publication
+ * @snippet tutorial.c Subscribing
+ *
+ * Publications are received asynchronously.  The first step in
+ * receiving a publication is to provide the publication handler to
+ * DPS and start the subscription.  The publication handler will be
+ * called for each received publication.
+ *
+ * @note Each instance of @c DPS_Node runs in its own thread.  The
+ * publication handler is dispatched from this thread.
+ *
+ * @snippet tutorial.c Receiving a publication
+ *
+ * This publication handler exercises the APIs for retrieving the
+ * subscription and publication information.
+ *
+ * @section acknowledging Acknowledging
+ *
+ * Acknowledgements provide a means for subscribers to reply to
+ * publications.  Similar to publications, acknowledgements may
+ * include an application payload, and no assumptions are made by DPS
+ * with regards to the acknowledgement payload format.
+ *
+ * @subsection requesting-an-acknowledgement Requesting an acknowledgement
+ * @snippet tutorial.c Requesting an acknowledgement
+ *
+ * Requesting an acknowledgement is identical to @ref creating-a-publication,
+ * with the addition of the @c DPS_AcknowledgementHandler.
+ *
+ * @subsection sending-an-acknowledgement Sending an acknowledgement
+ * @snippet tutorial.c Sending an acknowledgement
+ *
+ * To determine if a publication has requested an ack, call
+ * DPS_PublicationIsAckRequested().  To send an acknowledgement, along
+ * with any optional acknowledgement payload, call
+ * DPS_AckPublication().
+ *
+ * The @c pub parameter of the publication handler is only valid
+ * during the body of the handler.  In order to acknowledge a
+ * publication after the handler has returned, the application must
+ * first call DPS_CopyPublication() to create a partial copy of the
+ * publication.  The copy may be used after the handler returns.
+ *
+ * @subsection receiving-an-acknowledgement Receiving an acknowledgement
+ * @snippet tutorial.c Receiving an acknowledgement
+ *
+ * Acknowledgements are received asynchronously.  The acknowledgement
+ * handler will be called for each received acknowledgement.
+ *
+ * This acknowledgement handler exercises the APIs for retrieving the
+ * publication information associated with the acknowledgement.
+ *
+ * @note The acknowledgement handler is dispatched from the @c
+ * DPS_Node's thread.
+ */
```

Comment:
I feel like there is something fundamental I am missing about the ack handler but I can not put it into words right now.  Something to do with knowing who is acknowledging? Does the publisher have a local list of all the subscribers? Can the publisher mark off all the subscribers that sent an ack? This is me mostly thinking as I read the document. You can probably ignore this comment.

Reply:
I'll try and clear this up in the description of the acknowledgements

Reply:
At the DPS protocol (and API) layer publishers know nothing about subscribers and visa versa. If a publisher is expecting to hear from a specific set of subscribers an application-specific identifier will need to be included in the payload.